hdfs3/tests/test_hdfs3.py

```diff
@@ -635,3 +626,39 @@ def test_chown(hdfs):
     hdfs.touch(a)
     i = hdfs.info(a)
     hdfs.chown(a, 'root', 'supergroup')
+
+
+def test_text_bytes(hdfs):
+    with pytest.raises(NotImplementedError):
+        hdfs.open(a, 'wt')
+
+    with pytest.raises(NotImplementedError):
+        hdfs.open(a, 'rt')
+
+    with hdfs.open(a, 'wb') as f:
+        f.write(b'123')
+
+    with hdfs.open(a, 'rb') as f:
+        b = f.read()
+
+    assert b == b'123'
+
+
+def test_open_deep_file(hdfs):
+    with pytest.raises(IOError):
+        f = hdfs.open('/tmp/test/a/b/c/d/e/f', 'wb')
+
+
+def test_append(hdfs):
+    with hdfs.open(a, mode='ab') as f:
+        f.write(b'123')
+    with hdfs.open(a, mode='ab') as f:
+        f.write(b'456')
+
+    with hdfs.open(a, mode='rb') as f:
+        assert f.read() == b'123456'
+
+    with hdfs.open(a, mode='ab') as f:
+        f.write(b'789')
+    with hdfs.open(a, mode='rb') as f:
+        assert f.read() == b'123456789'
```

Comment:
OK, I plan to turn off appends for now, raise an issue, and leave this as an xfailed test.

If you see an easy fix please jump in here, either today or sometime in the indeterminate future.